projects/components/table/src/subcomponents/table-pagination.component.spec.ts

```diff
@@ -1,44 +1,140 @@
+  searchLabel: 'f',
+  sortLabel: 'g',
+  refreshListLabel: 'h',
+  settingsLabel: 'i',
+  noEntriesLabel: 'j',
+  saveLabel: 'k',
+  cancelLabel: 'l',
+  displayedColumnsLabel: 'm',
+};
 
-    spyOn(paginator.page, 'emit');
+@Component({
+  template: `
+    <ps-table-pagination
+      [pageSize]="pageSize"
+      [dataLength]="dataLength"
+      [pageIndex]="pageIndex"
+      [pageSizeOptions]="pageSizeOptions"
+      [intl]="intl"
+      [pageDebounce]="pageDebounce"
+      (page)="onPage($event)"
+    ></ps-table-pagination>
+  `,
+})
+class PaginationTestComponent {
+  public pageSize = 5;
+  public dataLength = 15;
+  public pageIndex = 0;
+  public intl = testIntl;
+  public pageDebounce: number;
 
-    paginator.onPage({ length: 9999 } as PageEvent);
+  @ViewChild(PsTablePaginationComponent, { static: false })
+  public pagination: PsTablePaginationComponent;
 
-    expect(paginator.page.emit).not.toHaveBeenCalled();
-    tick(301);
-    expect(paginator.page.emit).toHaveBeenCalledTimes(1);
-    expect(paginator.page.emit).toHaveBeenCalledWith({ length: 9999 } as PageEvent);
-  }));
+  public onPage = (_: PageEvent) => {};
+}
 
-  it('should have the correct amount of pages', fakeAsync(() => {
-    const paginator = createPaginationInstance();
-    paginator.ngOnInit();
-    spyOn(paginator.dataSource._internalDetectChanges, 'next').and.callThrough();
-    paginator.dataSource.updateData();
+fdescribe('PsTablePaginationComponent', () => {
+  describe('Unit tests', () => {
+    let component: PsTablePaginationComponent;
+    const cd = { markForCheck: () => {} } as ChangeDetectorRef;
+    beforeEach(() => {
+      component = new PsTablePaginationComponent(cd);
+    });
 
-    expect(paginator.dataSource._internalDetectChanges.next).toHaveBeenCalledWith();
-    expect(paginator.dataSource.pages).toEqual(3);
-    expect(paginator.pages.length).toBe(3);
-    expect(paginator.pages).toEqual([1, 2, 3]);
-  }));
-});
+    it('should set Intl correctly', () => {
+      component.intl = testIntl;
+      component.paginator = new MatPaginator(new MatPaginatorIntl(), cd);
 
-function createPaginationInstance(): PsTablePaginationComponent {
-  const cd: any = { markForCheck: () => {} };
-  const pagination = new PsTablePaginationComponent(cd);
-  pagination.dataSource = new PsTableDataSource(
-    () => of(Array.from({ length: 50 }, (_, i: number) => ({ id: i, str: `item ${i}` }))),
-    'client'
-  );
+      expect(component.paginator._intl.firstPageLabel).toEqual('a');
+      expect(component.paginator._intl.lastPageLabel).toEqual('b');
+      expect(component.paginator._intl.previousPageLabel).toEqual('c');
+    });
 
-  return pagination;
-}
+    it('should debounce pageEvent, if pageDebounce-Property is set', fakeAsync(() => {
+      const pageEmitSpy = spyOn(component.page, 'emit');
+      component.pageDebounce = 300;
+      component.onPage({ length: 9999 } as PageEvent);
+
+      expect(pageEmitSpy).not.toHaveBeenCalled();

```

Comment:
you should make this check after tick(299), to make sure it debounces for the whole duration